x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/chart_switch/chart_option_append.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import './chart_switch.scss';
+import React, { ReactNode } from 'react';
+import { EuiFlexItem, EuiIconTip, EuiBetaBadge, EuiFlexGroup } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+
+export const getDataLossWarning = (dataLoss: 'nothing' | 'layers' | 'everything' | 'columns') => {
+  if (dataLoss === 'nothing') {
+    return;
+  }
+  if (dataLoss === 'everything') {
+    return i18n.translate('xpack.lens.chartSwitch.dataLossEverything', {
+      defaultMessage: 'Changing to this visualization clears the current configuration.',
+    });
+  }
+  if (dataLoss === 'layers') {
+    return i18n.translate('xpack.lens.chartSwitch.dataLossLayersDescription', {
+      defaultMessage:
+        'Changing to this visualization modifies currently selected layer`s configuration and removes all other layers.',
+    });
+  } else
+    return i18n.translate('xpack.lens.chartSwitch.dataLossColumns', {
+      defaultMessage: `Changing to this visualization modifies the current configuration.`,
+    });
+};
+
+const DataLossWarning = ({ content, id }: { content: ReactNode; id: string }) => {
+  if (!content) return null;
```

Comment:
When content can be undefined?

Reply:
It can be null! In case if the dataLoss is `nothing` (we don't show warning)